classes/services/PKPFileService.inc.php

```diff
@@ -117,22 +117,35 @@ public function delete($id) {
 	 * This method sends a HTTP response and ends the request handling.
 	 * No code will run after this method is called.
 	 *
-	 * @param string $path The path to the file
+	 * @param string|int $pathOrFileId The path to the file or file ID
 	 * @param string $filename Filename to give to the downloaded file
 	 * @param boolean $inline Whether to stream the file to the browser
 	 */
-	public function download($path, $filename, $inline = false) {
+	public function download($pathOrFileId, $filename, $inline = false) {
+
+		$dispatcher = Application::get()->getRequest()->getDispatcher();
+
+		if (is_integer($pathOrFileId)) {
```

Comment:
Use `is_int` instead. Just for consistency.

Reply:
fixed